src/main/scala/dsptools/numbers/chisel_types/SIntTypeClass.scala

```diff
@@ -130,7 +142,7 @@ trait SIntInteger extends SIntRing with SIntIsReal with ConvertableToSInt with
   // fromSInt also included in Ring
   override def fromInt(n: Int): SInt = super[ConvertableToSInt].fromInt(n)
   // Overflow only on most negative
-  def abs(a: SInt): SInt = Mux(isSignNegative(a), super[SIntRing].minus(0.S, a), a)
+  def abs(a: SInt): SInt = Mux(isSignNegative(a), super[SIntRing].minusContext(0.S, a), a)
```

Comment:
Why is abs using minusContext? 

abs should use the correct one; 

abs_context should use minusContext.